Lib/defcon/objects/layer.py

```diff
@@ -251,7 +251,7 @@ def _insertGlyph(self, glyph, beginObservations=True):
     # -------------
 
     def __iter__(self):
```

Comment:
this `__iter__` is unnecessarily complex.

It could just be:

```python
def __iter__(self):
    for name in self.keys():
        yield self[name]
```

also, the comment says "Dict Behavior" but real dicts don't behave like that. Their `__iter__` yields keys, not values.
This is something from Robofab that confused me when I was learning python.